### ISSUE TITLE
What is the status of this project?

### ISSUE THREAD

**Issue by sintaloo** [JoseExposito/touchegg#334]
I am sorry about this question.
I failed to find any useful information from README.md and issues list. 
The last commit of this project is at 2017.
And many people complain it's not working in the recent linux releases.
I am wondering what the status of this project is. 
Thanks a lot.

**Comment by JoseExposito** [comment]
Hi @sintaloo 

The linux desktop stack has changed a lot since I started this project. Unfortunately, making impossible to use Touchegg anymore. Let me explain it a little bit:

uTouch:

uTouch is a library developed by Canonical to add multitouch support in Unity. Touchegg uses under the hood this library. Unfortunately the library is not maintained anymore and it is not even possible to use it in Wayland based desktop environments.

Wayland:

Wayland **compositors** access hardware input (including multitouch input) using libinput.
The _problem_ is that only compositors can have access to this library and not third party tools like Touchegg.

Long story sort, your desktop environment (Gnome, KDE...) is most likely running on Wayland, so Touchegg will not work :(
In a modern desktop stack your desktop environment is in charge of adding multitouch support and unfortunately there is nothing I can do about it.

**TL;DR** In conclusion and asking to your question, the status of this project is: If you are using an old distro, it will work. However it is not possible for me to migrate it so it works in new distros.

**Comment by sintaloo** [comment]
Dear @JoseExposito ,
Thank you very much for this detail explanation. 
Btw, I don't think modern distributions use Wayland by default. I am running debian 10 with XFCE, which uses X11. I checked the latest Ubuntu 20.04 which uses X11 by default too.
I haven't tried touchegg on debian 10 with XFCE since it's not in the repo and the project here is not updated since 2017. So I am not sure if it will work or not.
I did google on touchscreen articles and it seems touchegg is the only solution in Linux to enable normal gestures on a touch screen. I don't know what to do if touchegg can not be used.
OK, thanks a lot for your reply again. I am OK to close this issue. I saw you pinned this issue so I am not sure what you want to do with it. 
Cheers.

**Comment by thiswillbeyourgithub** [comment]
@JoseExposito  Thanks a lot for this answer.

I am using i3 without compositor and was thinking about adding a compositor (like compton or the like). Could you give any hint as to how to make touchegg work with i3 ?

Thanks!

**Comment by m-terlinde** [comment]
Maybe https://github.com/iberianpig/fusuma is useful for you.

**Comment by thiswillbeyourgithub** [comment]
could be, thanks a lot!

**Comment by m-terlinde** [comment]
I suggest to close this issue, @sintaloo.

**Comment by JoseExposito** [comment]
Hi all!

First at all, thank you very much for your feedback.

I created and pinned an issue to explain the status of the project:

https://github.com/JoseExposito/touchegg/issues/335

I'm working on a new version of Touchégg with libinput support, so you can expect to be able to run it in your laptop sooner rather than later.

I'll be updating that issue (and my Twitter account) with the progress of the new version.
